packages/services/src/rewards.service.ts

```diff
@@ -110,18 +139,24 @@ export class RewardsService {
         boostAllocationWindow,
       );
 
-      // TODO: add user_id, and agent_id columns to rewards table
-      // so we can track the rewards for each user and agent
-      await this.rewardsRepo.insertRewards(
-        rewards.map((reward) => ({
-          competitionId: competitionId,
-          address: reward.address,
-          amount: reward.amount,
-          leafHash: hexToBytes(
-            createLeafNode(reward.address as Hex, reward.amount),
-          ),
-          id: crypto.randomUUID(),
-        })),
+      const rewardsToInsert = rewards.map((reward) => ({
+        userId: reward.owner,
+        agentId: reward.competitor,
+        competitionId: competitionId,
+        address: reward.address,
+        amount: reward.amount,
+        leafHash: hexToBytes(
+          createLeafNode(reward.address as Hex, reward.amount),
+        ),
+        id: crypto.randomUUID(),
+      }));
+      await runWithConcurrencyLimit(
```

Comment:
made a change to run the inserts concurrently using `runWithConcurrencyLimit` helper